### PR TITLE
ci: Remove relaxed mode from eslint

### PR DESCRIPTION
**Pull request by ryan953**
Quick update to remove `SENTRY_ESLINT_RELAXED` from the eslint config, and make that file look a little more similar to the getsentry version.

The GETSENTRY version of this is https://github.com/getsentry/getsentry/pull/13274

also i took the chance to update prettier, since it runs under eslint, to make sure that we're treating all js & ts files the same. Treating files the same is a followup of https://github.com/getsentry/sentry/pull/66837

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,7 +1,6 @@
 **/dist/**/*
 **/vendor/**/*
 **/tests/**/fixtures/**/*
-/scripts/*.js
```
Comment (ryan953) [inline .eslintignore:4]: There are two js files in here, lets just lint them

File: package.json [hunk 2]
```diff
@@ -227,11 +227,11 @@
     "lint:js": "eslint static/app tests/js --ext .js,.jsx,.ts,.tsx",
     "lint:css": "stylelint 'static/app/**/*.[jt]sx'",
     "lint:biome": "biome check .",
-    "lint:prettier": "prettier \"**/*.md\" \"**/*.yaml\" \"**/*.{ts,tsx}\" --check",
+    "lint:prettier": "prettier \"**/*.md\" \"**/*.yaml\" \"**/*.[jt]s?(x)\" --check",
```
Comment (ryan953) [inline package.json:230]: we need to treat js,jsx,ts,tsx files equally.

File: .prettierignore
Comment (ryan953) [inline .prettierignore:1]: these js files/fixtures, now added to prettier, need to be ignored. there are some "json" payloads in js files, and minified files too.

File: package.json (hunk 2 above)
Comment (scttcper) [inline package.json:230]: i think you want this with the x being optional. might just be easier to do `{js,ts,jsx,tsx}`
```suggestion
    "lint:prettier": "prettier \"**/*.md\" \"**/*.yaml\" \"**/*.[jt]s(x)?\" --check",
```

Comment (ryan953) [reply package.json:230]: i had `{js,ts,jsx,tsx}` but it was complaining in one of the repos that no files were found. So i wanted to do the confusing regexp thing to keep the repos the same, and get full coverage if a new file (like a script) pops up

**Review by scttcper — APPROVED** [review]

**Comment by ryan953** [comment]
reverted because this failed in CI on master, but the branch passes locally and obvs the PR passed in CI too

SCR-20240313-owoc
SCR-20240313-owqm

**Pull request merged**